Repository: plone.app.users


Branch: refs/heads/master
Date: 2019-10-11T16:10:31+02:00
Author: Thomas Desvenain (tdesvenain) <thomas.desvenain@gmail.com>
Commit: https://github.com/plone/plone.app.users/commit/54328df9db492924255a31d576cae7b4575d3c5a

Error happening during user add notification is logged

fixes #85

Files changed:
A news/85.enhancement
M plone/app/users/browser/register.py

b"diff --git a/news/85.enhancement b/news/85.enhancement\nnew file mode 100644\nindex 0000000..f2ebe41\n--- /dev/null\n+++ b/news/85.enhancement\n@@ -0,0 +1 @@\n+Error happening during user add notification is logged. fixes #85\ndiff --git a/plone/app/users/browser/register.py b/plone/app/users/browser/register.py\nindex 63ca305..3ece5e7 100644\n--- a/plone/app/users/browser/register.py\n+++ b/plone/app/users/browser/register.py\n@@ -479,7 +479,8 @@ def handle_join_success(self, data):\n             except ConflictError:\n                 # Let Zope handle this exception.\n                 raise\n-            except Exception:\n+            except Exception as err:\n+                logging.exception(err)\n                 ctrlOverview = getMultiAdapter((portal, self.request),\n                                                name='overview-controlpanel')\n                 mail_settings_correct = not ctrlOverview.mailhost_warning()\n"

Repository: plone.app.users


Branch: refs/heads/master
Date: 2019-10-11T16:11:53+02:00
Author: Thomas Desvenain (tdesvenain) <thomas.desvenain@gmail.com>
Commit: https://github.com/plone/plone.app.users/commit/87d7fa55a15458ded20380baf701a178ebfe09d1

User feedback if error happens during user add notification. fixes #86

Files changed:
A news/86.enhancement
M plone/app/users/browser/register.py

b'diff --git a/news/86.enhancement b/news/86.enhancement\nnew file mode 100644\nindex 0000000..105e8f0\n--- /dev/null\n+++ b/news/86.enhancement\n@@ -0,0 +1 @@\n+User feedback if error happens during user add notification. fixes #86\ndiff --git a/plone/app/users/browser/register.py b/plone/app/users/browser/register.py\nindex 3ece5e7..819c973 100644\n--- a/plone/app/users/browser/register.py\n+++ b/plone/app/users/browser/register.py\n@@ -401,8 +401,6 @@ def action_join(self, action):\n \n         self.handle_join_success(data)\n \n-        self._finishedRegister = True\n-\n         # XXX Return somewhere else, depending on what\n         # handle_join_success returns?\n         came_from = self.request.form.get(\'came_from\')\n@@ -450,6 +448,7 @@ def handle_join_success(self, data):\n         except (AttributeError, ValueError) as err:\n             logging.exception(err)\n             IStatusMessage(self.request).addStatusMessage(err, type="error")\n+            self._finishedRegister = False\n             return\n \n         if user_id != login_name:\n@@ -462,6 +461,7 @@ def handle_join_success(self, data):\n         self.applyProperties(user_id, data)\n \n         settings = self._get_security_settings()\n+        self._finishedRegister = True\n         if data.get(\'mail_me\') or (not settings.enable_user_pwd_choice and\n                                    not data.get(\'password\')):\n             # We want to validate the email address (users cannot\n@@ -491,6 +491,7 @@ def handle_join_success(self, data):\n                     # Remove the account:\n                     self.context.acl_users.userFolderDelUsers(\n                         [user_id], REQUEST=self.request)\n+                    self._finishedRegister = False\n                     IStatusMessage(self.request).addStatusMessage(\n                         _(u\'status_fatal_password_mail\',\n                           default=u"Failed to create your account: we were "\n@@ -692,6 +693,9 @@ def action_join(self, action):\n \n         self.handle_join_success(data)\n \n+        if not self._finishedRegister:\n+            return\n+\n         portal_groups = getToolByName(self.context, \'portal_groups\')\n         user_id = data[\'user_id\']\n         is_zope_manager = getSecurityManager().checkPermission(\n@@ -717,4 +721,3 @@ def action_join(self, action):\n             self.context.absolute_url() +\n             \'/@@usergroup-userprefs?searchstring=\' + user_id)\n \n-        self._finishedRegister = True\n'

Repository: plone.app.users


Branch: refs/heads/master
Date: 2019-10-20T09:58:51+02:00
Author: Jens W. Klein (jensens) <jk@kleinundpartner.at>
Commit: https://github.com/plone/plone.app.users/commit/02df6ebaf1775b7888a694393231fe5debf0e0fd

Merge pull request #87 from plone/fix_notification_error_feedback

Fix notification error feedback

Files changed:
A news/85.enhancement
A news/86.enhancement
M plone/app/users/browser/register.py

b'diff --git a/news/85.enhancement b/news/85.enhancement\nnew file mode 100644\nindex 0000000..f2ebe41\n--- /dev/null\n+++ b/news/85.enhancement\n@@ -0,0 +1 @@\n+Error happening during user add notification is logged. fixes #85\ndiff --git a/news/86.enhancement b/news/86.enhancement\nnew file mode 100644\nindex 0000000..105e8f0\n--- /dev/null\n+++ b/news/86.enhancement\n@@ -0,0 +1 @@\n+User feedback if error happens during user add notification. fixes #86\ndiff --git a/plone/app/users/browser/register.py b/plone/app/users/browser/register.py\nindex 63ca305..819c973 100644\n--- a/plone/app/users/browser/register.py\n+++ b/plone/app/users/browser/register.py\n@@ -401,8 +401,6 @@ def action_join(self, action):\n \n         self.handle_join_success(data)\n \n-        self._finishedRegister = True\n-\n         # XXX Return somewhere else, depending on what\n         # handle_join_success returns?\n         came_from = self.request.form.get(\'came_from\')\n@@ -450,6 +448,7 @@ def handle_join_success(self, data):\n         except (AttributeError, ValueError) as err:\n             logging.exception(err)\n             IStatusMessage(self.request).addStatusMessage(err, type="error")\n+            self._finishedRegister = False\n             return\n \n         if user_id != login_name:\n@@ -462,6 +461,7 @@ def handle_join_success(self, data):\n         self.applyProperties(user_id, data)\n \n         settings = self._get_security_settings()\n+        self._finishedRegister = True\n         if data.get(\'mail_me\') or (not settings.enable_user_pwd_choice and\n                                    not data.get(\'password\')):\n             # We want to validate the email address (users cannot\n@@ -479,7 +479,8 @@ def handle_join_success(self, data):\n             except ConflictError:\n                 # Let Zope handle this exception.\n                 raise\n-            except Exception:\n+            except Exception as err:\n+                logging.exception(err)\n                 ctrlOverview = getMultiAdapter((portal, self.request),\n                                                name=\'overview-controlpanel\')\n                 mail_settings_correct = not ctrlOverview.mailhost_warning()\n@@ -490,6 +491,7 @@ def handle_join_success(self, data):\n                     # Remove the account:\n                     self.context.acl_users.userFolderDelUsers(\n                         [user_id], REQUEST=self.request)\n+                    self._finishedRegister = False\n                     IStatusMessage(self.request).addStatusMessage(\n                         _(u\'status_fatal_password_mail\',\n                           default=u"Failed to create your account: we were "\n@@ -691,6 +693,9 @@ def action_join(self, action):\n \n         self.handle_join_success(data)\n \n+        if not self._finishedRegister:\n+            return\n+\n         portal_groups = getToolByName(self.context, \'portal_groups\')\n         user_id = data[\'user_id\']\n         is_zope_manager = getSecurityManager().checkPermission(\n@@ -716,4 +721,3 @@ def action_join(self, action):\n             self.context.absolute_url() +\n             \'/@@usergroup-userprefs?searchstring=\' + user_id)\n \n-        self._finishedRegister = True\n'

